client/recordUtil.js

```diff
@@ -301,13 +303,24 @@ module.exports.syncRecordAsJS = (record) => {
    * .toObject() options:
    * http://dcode.io/protobuf.js/global.html#ConversionOptions
    */
-  let object = record.toObject()
+  const object = api.SyncRecord.toObject(record)
   object.action = record.action
   const type = serializer.getSyncRecordObjectData(record)
+  if (!syncTypes[type]) {
+    throw new Error('Unknown sync objectData type: ' + type)
+  }
   object.objectData = type
-  const data = record[type].toObject({defaults: true, enums: Number, longs: Number})
+  const data = api.SyncRecord[syncTypes[type]].toObject(record[type], {
+    defaults: true,
+    enums: Number,
+    longs: Number
+  })
   if (data.fields && data.fields.length > 0) {
     object[type] = pickFields(data, data.fields)
+    // Empty the fields field for backwards compatibility with protobufjs 6.6
```

Comment:
i don't remember the details. what i do remember is pbjs didn't distinguish between props with unset default value versus fields explicitly set with that default value

so I added fields to explicitly track which fields were set.